Sources/TootSDK/Models/TootFeature.swift

```diff
@@ -1,8 +1,303 @@
+            FlavourRequirement(flavour: flavour, minDisplayVersion: nil, maxDisplayVersion: nil, minVersion: version, maxVersion: maxVersion)
+        }
+
+        /// Create a requirement for a maximum API version
+        /// - Parameters:
+        ///   - flavour: The server flavour
+        ///   - version: The maximum API version supported
+        public static func until(_ flavour: TootSDKFlavour, version: Int) -> FlavourRequirement {
+            FlavourRequirement(flavour: flavour, minDisplayVersion: nil, maxDisplayVersion: nil, minVersion: nil, maxVersion: version)
+        }
+
+        // MARK: - Display Version Requirements (Fallback/Legacy)
+
+        /// Create a requirement for a minimum display version
+        /// - Parameters:
+        ///   - flavour: The server flavour
+        ///   - displayVersion: The minimum display version required
+        public static func from(_ flavour: TootSDKFlavour, displayVersion: String) -> FlavourRequirement {
+            FlavourRequirement(
+                flavour: flavour, minDisplayVersion: Version(tolerant: displayVersion), maxDisplayVersion: nil, minVersion: nil, maxVersion: nil)
+        }
+
+        /// Create a requirement for a display version range
+        /// - Parameters:
+        ///   - flavour: The server flavour
+        ///   - displayVersion: The minimum display version required
+        ///   - to: The maximum display version supported
+        public static func from(_ flavour: TootSDKFlavour, displayVersion: String, to maxDisplayVersion: String) -> FlavourRequirement {
+            FlavourRequirement(
+                flavour: flavour,
+                minDisplayVersion: Version(tolerant: displayVersion),
+                maxDisplayVersion: Version(tolerant: maxDisplayVersion),
+                minVersion: nil,
+                maxVersion: nil
+            )
+        }
+
+        /// Create a requirement for a maximum display version
+        /// - Parameters:
+        ///   - flavour: The server flavour
+        ///   - displayVersion: The maximum display version supported
+        public static func until(_ flavour: TootSDKFlavour, displayVersion: String) -> FlavourRequirement {
+            FlavourRequirement(
+                flavour: flavour, minDisplayVersion: nil, maxDisplayVersion: Version(tolerant: displayVersion), minVersion: nil, maxVersion: nil)
+        }
+
+        // MARK: - Combined Requirements
+
+        /// Create a requirement with both API version and display version fallback
+        /// - Parameters:
+        ///   - flavour: The server flavour
+        ///   - version: The minimum API version required
+        ///   - fallbackDisplayVersion: The minimum display version to use as fallback when API version is not available
+        public static func from(_ flavour: TootSDKFlavour, version: Int, fallbackDisplayVersion: String) -> FlavourRequirement {
```

Comment:
This one can be merged with one of previous by making fallback parameter optional. Less autocompletion suggestions/API to cover that way.